contracts/utils/cryptography/P256.sol

```diff
@@ -183,6 +183,9 @@ library P256 {
     /**
      * @dev Point addition on the jacobian coordinates
      * Reference: https://www.hyperelliptic.org/EFD/g1p/auto-shortw-jacobian.html#addition-add-1998-cmo-2
+     *
+     * Note that if both input points are identical, this function will return `(0, 0, 0)` instead of doubling the point.
+     * The probability of this happening naturally is negligible (approximately 2048 / (2^256 - 1), or about 10^-74).
```

Comment:
If woth input are identical, the probability of this happening is 100%.

What that probability here ? If the the probability of two random points being identical ? Is it something else ?

```suggestion
     * The probability of this happening naturally is negligible (approximately 2048 / (2^256 - 1), or about 10^-74).
```